localstack/services/cloudformation/engine/template_preparer.py

```diff
@@ -29,21 +37,156 @@ def template_to_json(template: str) -> str:
+    region_before = os.environ.get("AWS_DEFAULT_REGION")
+    if boto3.session.Session().region_name is None:
+        os.environ["AWS_DEFAULT_REGION"] = aws_stack.get_region()
+    loader = create_policy_loader()
 
+    try:
+        transformed = transform_sam(parsed_template, {}, loader)
+        return transformed
+    except Exception as e:
+        raise FailedTransformation(transformation=SERVERLESS_TRANSFORM, message=str(e))
+    finally:
         # Note: we need to fix boto3 region, otherwise AWS SAM transformer fails
-        region_before = os.environ.get("AWS_DEFAULT_REGION")
-        if boto3.session.Session().region_name is None:
-            os.environ["AWS_DEFAULT_REGION"] = aws_stack.get_region()
-        loader = create_policy_loader()
-        try:
-            transformed = transform_sam(parsed, {}, loader)
-            return json.dumps(transformed)
-        finally:
-            os.environ.pop("AWS_DEFAULT_REGION", None)
-            if region_before is not None:
-                os.environ["AWS_DEFAULT_REGION"] = region_before
+        os.environ.pop("AWS_DEFAULT_REGION", None)
+        if region_before is not None:
+            os.environ["AWS_DEFAULT_REGION"] = region_before
+
+
+def format_transforms(transforms: List | Dict | str) -> List[Dict]:
+    formatted_transformations = []
+    if isinstance(transforms, str):
+        formatted_transformations.append({"Name": transforms})
+
+    if isinstance(transforms, Dict):
+        formatted_transformations.append(transforms)
+
+    if isinstance(transforms, list):
+        for transformation in transforms:
+            if isinstance(transformation, str):
+                formatted_transformations.append({"Name": transformation})
+            if isinstance(transformation, Dict):
+                formatted_transformations.append(transformation)
+
+    return formatted_transformations
+
+
+def resolve_parameters(template_parameters: dict, request_parameters: List[dict]):
+    result = {}
+    # add default template parameter values
+    for key, value in template_parameters.items():
+        param_value = value.get("Default")
+        result[key] = {
+            "ParameterKey": key,
+            "ParameterValue": param_value,
+        }
+        param_type = value.get("Type", "")
+        if not param_type:
+            if param_type == "AWS::SSM::Parameter::Value<String>":
+                result[key]["ResolvedValue"] = resolve_ssm_parameter_value(param_type, param_value)
+            elif param_type.startswith("AWS::"):
+                LOG.info(
+                    f"Parameter Type '{param_type}' is currently not supported. Coming soon, stay tuned!"
+                )
+            else:
+                # lets assume we support the normal CFn parameters
+                pass
+
+    # add stack parameters
+    result.update({p["ParameterKey"]: p for p in request_parameters})
+    result = list(result.values())
+    return result
```

Comment:
nit: please add some context here (e.g. why this was partially copied over from the deployer)